frontend/packages/dev-console/src/components/pipelines/utils/__tests__/pipeline-operator.spec.ts

```diff
@@ -0,0 +1,111 @@
+const k8sListMock = k8sList as jest.Mock;
+
+describe('getPipelineOperatorVersion', () => {
+  it('should fetch the ClusterServiceVersion from the api', async () => {
+    const csvs = [
+      {
+        metadata: { name: 'openshift-pipelines-operator.v1.0.1' },
+        spec: { version: '1.0.1' },
+        status: { phase: 'Succeeded' },
+      } as ClusterServiceVersionKind,
+    ];
+    k8sListMock.mockReturnValueOnce(Promise.resolve(csvs));
+
+    const version = await getPipelineOperatorVersion('unit-test');
+
+    expect(version.raw).toBe('1.0.1');
+    expect(version.major).toBe(1);
+    expect(version.minor).toBe(0);
+    expect(version.patch).toBe(1);
+    expect(k8sList).toHaveBeenCalledTimes(1);
+  });
+
+  it('should return the active ClusterServiceVersion if multiple returns', async () => {
+    const csvs = [
+      {
+        metadata: { name: 'openshift-pipelines-operator.v1.0.1' },
+        spec: { version: '1.0.1' },
+        status: { phase: 'Succeeded' },
+      } as ClusterServiceVersionKind,
+      {
+        metadata: { name: 'openshift-pipelines-operator.v1.1.1' },
+        spec: { version: '1.1.1' },
+        status: { phase: 'Pending' },
+      } as ClusterServiceVersionKind,
+    ];
+    k8sListMock.mockReturnValueOnce(Promise.resolve(csvs));
+
+    const version = await getPipelineOperatorVersion('unit-test');
+
+    expect(version.raw).toBe('1.0.1');
+    expect(version.major).toBe(1);
+    expect(version.minor).toBe(0);
+    expect(version.patch).toBe(1);
+    expect(k8sList).toHaveBeenCalledTimes(1);
+  });
+
+  it('should fetch the latest (highest) ClusterServiceVersion from the api', async () => {
+    const csvs = [
+      {
+        metadata: { name: 'openshift-pipelines-operator.v1.0.1' },
+        spec: { version: '1.0.1' },
+        status: { phase: 'Succeeded' },
+      } as ClusterServiceVersionKind,
+      {
+        metadata: { name: 'openshift-pipelines-operator.v10.11.12' },
+        spec: { version: '10.11.12' },
+        status: { phase: 'Succeeded' },
+      } as ClusterServiceVersionKind,
+      {
+        metadata: { name: 'openshift-pipelines-operator.v1.1.1' },
+        spec: { version: '1.1.1' },
+        status: { phase: 'Succeeded' },
+      } as ClusterServiceVersionKind,
+    ];
+    k8sListMock.mockReturnValueOnce(Promise.resolve(csvs));
+
+    const version = await getPipelineOperatorVersion('unit-test');
+
+    expect(version.raw).toBe('10.11.12');
+    expect(version.major).toBe(10);
+    expect(version.minor).toBe(11);
+    expect(version.patch).toBe(12);
+    expect(k8sList).toHaveBeenCalledTimes(1);
+  });
+
+  it('should return null if there is no ClusterServiceVersion available', async () => {
+    k8sListMock.mockReturnValueOnce(Promise.resolve([]));
+    await expect(getPipelineOperatorVersion('unit-test')).resolves.toBe(null);
+    expect(k8sList).toHaveBeenCalledTimes(1);
+  });
+
+  it('should return null if there is no matching ClusterServiceVersion available', async () => {
```

Comment:
Could you write one more test for valid items but not including a Pipeline Operator? I think you covered both sides (none and mixed), but I think there is still valid test possible here for just invalid ones.

Reply:
added new test, it should return null if there is no pipeline operator at all